micro-rdk-nmea/src/lib.rs

```diff
@@ -26,11 +29,10 @@ mod tests {
         let mut data = Vec::<u8>::new();
         let res = general_purpose::STANDARD.decode_vec(water_depth_str, &mut data);
         assert!(res.is_ok());
-        let cursor = DataCursor::new(data[33..].to_vec());
-        let message = WaterDepth::from_cursor(cursor, 13);
+        let cursor = DataCursor::new(data[32..].to_vec());
```

Comment:
So, before, what was happening by starting at byte 33?

Also, is there a reason in this test and the others not to interpret the metadata in the 32 bytes at the front? It seems worthwhile to test that it parses correctly.

Finally, it looks like `WaterDepth` still does have a `source_id`, so, I'm curious why the value is no longer being validated.

Reply:
Byte 32 was the source_id for those messages that have one, but not every message has a source_id field. I can add checks in the tests for the ones that do

Reply:
I think it is worth checking the source_id is decoded correctly for those types that have them.

I also think it is worth parsing the metadata here and validating the decoded values.

Reply:
Thanks for adding the check of the `source_id`. Did you disagree with the idea to validate the metadata too?

Reply:
I don't feel too strongly about it. I was suggesting it mostly because you already have static test data here, but the first 32 bytes get thrown away, when it could be a perfectly good test of decoding the metadata. If you do a separate test then you would just need to obtain the metadata bytes from somewhere anyway. Up to you.

Reply:
wait I do have a unit test for this actually, does this really matter for every message type?

Reply:
No. Maybe just do one of them, if you want to.